### DISCUSSION TITLE
Does `with_these_medications` include "issued by hospital meds"

### DISCUSSION THREAD

**Discussion by brianmackenna** [opensafely/documentation#527] [Q&A]
The majority of routine medications are prescribed by GP and will be are available to query via `with_these_medications`. 

However people can receive medications (including routine ones) from other settings such as A&E, hospital outpatients, dentists, specialist mental health trusts. It can be important for GPs to know about these medications (for example interactions with routine medications) and al the primary care systems have the facility for medications prescribed elsewhere to be recorded in the patient record. This ensures that clinicians have a record to inform treatment and can trigger things like decision support. Brief description on how to record these medications in EMIS and TPP is [available here](https://digital.nhs.uk/services/summary-care-records-scr/recording-medicines-prescribed-elsewhere-into-the-gp-practice-record).

My questions is...................

 using `with_these_medications` if a medication prescribed elsewhere has been recorded in the GP record as per EMIS/TPP instructions can we identify them?

**Comment by iaindillingham** [comment]
Just to check my understanding: You wish to know whether [`with_these_medications`][1] can be used to identify the type of location where a given medication was prescribed? Normally, medications are prescribed by GPs, so the type of location may be "GP". However, they may also be prescribed by, for example, dentists, so the type of location may be "Dentist".

Based on reading the docs, I don't think the type of location is available through this function.

Looking at [the TPP backend][2], medication data are stored in the `MedicationIssue` table. I think this is linked to the `Consultation` table through `MedicationIssue.Consultation_ID`. Similarly, `Consultation` is linked to `RegistrationHistory`, which is linked to `Organisation`. I can't see a column on any of these tables that indicates the type of location: a column on the `Organisation` table would be a good candidate, but this table's columns contain mostly geographic information (STP, MSOA, region) rather than the type of location (e.g. "GP" or "Dentist"). `MedicationIssue.MedicationStatus` might be a good candidate, but it's unclear to me what the status codes (integers) indicate.

If the type of location were available in the TPP backend, then we would need to update cohort-extractor and Data Builder (cohort-extractor v2) to expose it.

[1]: https://docs.opensafely.org/study-def-variables/#cohortextractor.patients.with_these_medications
[2]: https://github.com/opensafely/database-notebooks/blob/master/notebooks/database-schema.ipynb

**Comment by evansd** [comment]
From a tech point of view all I can say is that this method queries the `MedicationIssue` table in TPP and the `medication_all_orgs_v2` table in EMIS. Exactly what Rx these tables contain is a question we'd need to raise with TPP and EMIS directly.

**Reply by sebbacon** [reply]
And then document it!
